### PR TITLE
Gds 498 tags

### PR DESCRIPTION
**Pull request by Lesz**
I've built and styled Tags - please have a look. I put last element as <a href> link so you can see :focus state. There's additional default right margin added to <button> elements by browser - in order to remove it I set up parent element "font-size: 0".

### CODE REVIEW

**Comment by seanelliott86** [comment]
@Lesz i would suggest we dont have a wrapper of `nsw-tags-list` because we are not sure if its going to be used in as a list, just use a class of `nsw-tags` on the buttons for now, lets keep it very minimal and without any assumptions on its usage just yet.

File: src/components/tags/_tags.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @include font-stack('body');
+    @include nsw-spacing(margin, none xs xs none);
+    @include nsw-spacing(padding, xs sm );
+    @include font-size('sm');
+    border: 2px solid $light40;
+    border-radius: $border-radius;
+    background-color: $white;
+  }
+
+  &__tag:hover{
```
Comment (seanelliott86) [inline src/components/tags/_tags.scss]: with SASS you can nest `:hover` and `:focus` inside the element that requires those states, saves you from having to write `&__tag:hover`

File: src/components/tags/_tags.hbs [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
-	<h1>Tags</h1>      
+	<h1>Tags</h1>
+    <div class="nsw-tags-list">
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">LongTag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag selected">Tag</button>
+      <a href="#" class="nsw-tags-list__tag">Link to show :focus state</a>
```
Comment (seanelliott86) [inline src/components/tags/_tags.hbs]: Remove the link, its not needed to show the focus state and we dont want to use  links for tags anyway.

File: src/components/tags/_tags.hbs [hunk 3]
```diff
@@ -1,3 +1,12 @@
 <div class="nsw-content nsw-container">
-	<h1>Tags</h1>      
+	<h1>Tags</h1>
+    <div class="nsw-tags-list">
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">LongTag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
+      <button type="button" class="nsw-tags-list__tag">Tag</button>
```
Comment (seanelliott86) [inline src/components/tags/_tags.hbs]: For each tag thats not selected add `aria-pressed="false"` this gives a clearer understanding to screen readers that this are not clicked yet

File: src/components/tags/_tags.hbs [hunk 4]
```diff
@@ -1,3 +1,12 @@
...
+      <button type="button" class="nsw-tags-list__tag selected">Tag</button>
```
Comment (seanelliott86) [inline src/components/tags/_tags.hbs]: Add `aria-pressed="true"` to the selected tags

File: src/components/tags/_tags.scss (hunk 1 above)
Comment (masonphanNSW) [reply src/components/tags/_tags.scss]: @seanelliott86 fixed in last commit

File: src/components/tags/_tags.hbs (hunk 3 above)
Comment (masonphanNSW) [reply src/components/tags/_tags.hbs]: @seanelliott86 fixed in last commit

File: src/components/tags/_tags.hbs (hunk 4 above)
Comment (masonphanNSW) [reply src/components/tags/_tags.hbs]: @seanelliott86 fixed in last commit

File: src/components/tags/_tags.hbs (hunk 2 above)
Comment (masonphanNSW) [reply src/components/tags/_tags.hbs]: @seanelliott86 fixed in last commit

**Pull request merged**